### PR TITLE
update 115

### PR DESCRIPTION
**Pull request by OpenVINO-dev-contest**
switch the curr_request and next_request to align with the pseudo code

### CODE REVIEW

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
Thanks @OpenVINO-dev-contest for these changes. To make it better I propose the following:
- "return sync_fps" and async_fps should be the last line in the function. Now you're returning before destroying windows in popup mode
- variables total and num should be more meaningful e.g. total_time, frame_number 
- these 3 lines are important only for displaying high-quality resolution streams, so please remove them from async and sync functions; it will simplify the code
```
scale = 1280 / max(frame.shape)
if scale < 1:
frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)
```
- it's probable that this update fix mac timeouts; please remove ignoring 115-async-api for CI in nbval.yml file

**Comment by OpenVINO-dev-contest** [comment]
> Thanks @OpenVINO-dev-contest for these changes. To make it better I propose the following:
> 
> * "return sync_fps" and async_fps should be the last line in the function. Now you're returning before destroying windows in popup mode
> * variables total and num should be more meaningful e.g. total_time, frame_number
> * these 3 lines are important only for displaying high-quality resolution streams, so please remove them from async and sync functions; it will simplify the code
> 
> ```
> scale = 1280 / max(frame.shape)
> if scale < 1:
> frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)
> ```
> 
> * there is a chance that this update fixes mac timeouts; to verify this please remove ignoring `115-async-api` for CI in `nbval.yml` file

hi @adrianboguszewski thanks for your suggestion. I have removed the codes you mentioned

File: .github/workflows/nbval.yml [hunk 1]
```diff
@@ -188,7 +188,7 @@ jobs:
       if: runner.os == 'MacOS'
       run: |
         chmod +x .ci/validate_notebooks.sh
-        .ci/validate_notebooks.sh 107-speech-recognition-quantization 114-quantization-simplified-mode 115-async-api
+        .ci/validate_notebooks.sh 107-speech-recognition-quantization 114-quantization-simplified-mode
```
Comment (adrianboguszewski) [inline .github/workflows/nbval.yml:191]: Unfortunately, it looks like this doesn't fix the bug for mac. So, we need to revert this change.

File: notebooks/115-async-api/115-async-api.ipynb [hunk 2]
```diff
@@ -324,28 +320,32 @@
...
     "            next_frame = player.next()\n",
-    "            if frame is None:\n",
+    "            if next_frame is None:\n",
     "                print(\"Source ended\")\n",
     "                break\n",
     "            # if frame larger than full HD, reduce size to improve the performance\n",
-    "            scale = 1280 / max(frame.shape)\n",
+    "            scale = 1280 / max(next_frame.shape)\n",
     "            if scale < 1:\n",
-    "                frame = cv2.resize(frame, None, fx=scale, fy=scale, interpolation=cv2.INTER_AREA)\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:341]: These 3 lines are not necessary

File: notebooks/115-async-api/115-async-api.ipynb [hunk 3]
```diff
@@ -324,28 +320,32 @@
...
+    "        resized_frame = preprocess(frame)\n",
+    "        curr_request.set_tensor(input_layer_ir, ov.Tensor(resized_frame))\n",
+    "        # Start the CURRENT inference request\n",
+    "        curr_request.start_async()\n",
     "        while True:\n",
-    "            # Read NEXT frame\n",
+    "            # Capture NEXT frame\n",
     "            next_frame = player.next()\n",
-    "            if frame is None:\n",
+    "            if next_frame is None:\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:335]: You can leave the name "frame" instead of the "next_frame" everywhere you use it. It would make code easier to understand.

File: notebooks/115-async-api/115-async-api.ipynb [hunk 4]
```diff
@@ -324,28 +320,32 @@
...
+    "            resized_frame = preprocess(next_frame)\n",
+    "            next_request.set_tensor(input_layer_ir, ov.Tensor(resized_frame))\n",
+    "            # Start the NEXT inference request\n",
+    "            next_request.start_async()\n",
+    "            # Waiting for CURRENT inference result\n",
+    "            if curr_request.wait_for(-1) == 1:\n",
+    "                res = curr_request.get_output_tensor(0).data\n",
     "                stop_time = time.time()\n",
     "                total = stop_time - start_time\n",
     "                num = num + 1\n",
```
Comment (adrianboguszewski) [inline notebooks/115-async-api/115-async-api.ipynb:351]: Variables total and num should be more meaningful e.g. total_time, frame_number. The same for sync function

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]

File: .github/workflows/nbval.yml (hunk 1 above)
Comment (adrianboguszewski) [reply .github/workflows/nbval.yml:191]: I've reported a bug in OpenVINO

File: notebooks/115-async-api/115-async-api.ipynb (hunk 3 above)
Comment (OpenVINO-dev-contest) [reply notebooks/115-async-api/115-async-api.ipynb:335]: Hi I dont think so, next_frame and frame are for 2 infer requests separately

Comment (OpenVINO-dev-contest) [reply notebooks/115-async-api/115-async-api.ipynb:335]: Hi I dont think so, next_frame and frame are for 2 infer requests separately

Comment (adrianboguszewski) [reply notebooks/115-async-api/115-async-api.ipynb:335]: You're only assigning the reference to the variable, you're not changing the value itself. Try it :)

**Review by adrianboguszewski — APPROVED** [review]

**Review by zhuo-yoyowz — APPROVED** [review]
